### DISCUSSION TITLE
Multi configuration

### DISCUSSION THREAD

**Discussion by DentOpt** [nasa/fprime#1638] [Q&A] [answered]
Hi, 

we are having two configurations that use the same project workspace:
```
MyProj
- MyComponent/
- TopologyFlight/
  - topology.fpp 
  - instances.fpp
  ...
- TopologySimulator/
  - topology.fpp
  - instances.fpp
  ...
- CMakeLists.txt
```

In MyProject/TopologyFlight/topology.cpp

```
module Macos {
   ....
    topology Macos {
        ...
```
In MyProject/Topologysimulatort/topology.cpp
```
module Macos {
   ....
    topology Simulator {
        ...
```

In the CMakeLists we want to create one executable for each configuration:
```
add_fprime_subdirectory("${CMAKE_CURRENT_LIST_DIR}/TopologyFlight/")
set(SOURCE_FILES "${CMAKE_CURRENT_LIST_DIR}/TopologyFlight/Main.cpp")
set(MOD_DEPS TopologyFlight)
register_fprime_deployment()   

add_fprime_subdirectory("${CMAKE_CURRENT_LIST_DIR}/TopologySimulator/")
set(SOURCE_FILES "${CMAKE_CURRENT_LIST_DIR}/TopologySimulator/Main.cpp")
set(MOD_DEPS TopologySimulator)
register_fprime_deployment()   
```
As we are using the same project workspace, the registering of the deployment collides: 

```
~MyProj/build-fprime-automatic-native/locs.fpp: 64.44
locate constant Macos.Default.queueSize at "../TopologySimulator/instances.fpp"
                                           ^
error: inconsistent location path ~MyProj/TopologySimulator/instances.fpp
~MyProj/build-fprime-automatic-native/locs.fpp: 63.44
locate constant Macos.Default.queueSize at "../TopologyFlight/instances.fpp"
                                           ^
previous path is ~MyProj/TopologyFlight/instances.fpp                         ^
```

Can you recommend the new standard workflow for this with v3.1.0 (using register_fprime_deployment)?

**Comment by Joshua-Anderson** [comment, marked as answer]
A deployment can only have a single topology, and there can only be one deployment to a directory.

What you're trying to do is actually a pretty common F' pattern. Just create a new top level directory for your `TopologySimulator` executable (instead of a second topology within the same folder) and everything should work like you expect.

You can always include components from other deployments in this deployment, allowing you to include components from the flight deployment within the simulation deployment.
